imports/pages/groups/profile/__tests__/Join.js

```diff
@@ -1,62 +1,104 @@
 import { shallow } from "enzyme";
 import { shallowToJson } from "enzyme-to-json";
-import Join from "../Join";
-
-const defaultProps = {
-  onClick: jest.fn(),
-  group: {
-    name: "happy group",
-    members: [
-      { role: "leader", person: { firstName: "jim" } },
-      { role: "leader", person: { nickName: "bob" } },
-    ],
-  },
-  onExit: jest.fn(),
-};
-
-const generateComponent = (additionalProps = {}) => {
-  const newProps = {
-    ...defaultProps,
-    ...additionalProps,
+import Join, { ShowPhoneTextArea } from "../Join";
+
+describe("Join", () => {
+  const defaultProps = {
+    onClick: jest.fn(),
+    group: {
+      name: "happy group",
+      members: [
+        { role: "leader", person: { firstName: "jim" } },
+        { role: "leader", person: { nickName: "bob" } },
+      ],
+    },
+    onExit: jest.fn(),
+    loading: false,
+    phones: [],
+  };
+
+  const generateComponent = (additionalProps = {}) => {
+    const newProps = {
+      ...defaultProps,
+      ...additionalProps,
+    };
+    return <Join { ...newProps } />;
   };
-  return <Join { ...newProps } />;
-};
-
-it("renders with props", () => {
-  const wrapper = shallow(generateComponent());
-  expect(shallowToJson(wrapper)).toMatchSnapshot();
-});
-
-it("onClick calls updates state and calls onClick", () => {
-  const mockOnClick = jest.fn();
-  const wrapper = shallow(generateComponent({
-    onClick: mockOnClick,
-  }));
-  wrapper.instance().onClick("e");
-  expect(wrapper.state().state).toBe("loading");
-  expect(mockOnClick).toHaveBeenCalledTimes(1);
-  expect(mockOnClick.mock.calls[0][0]).toBe("e");
-
-  // callback without error
-  const result = mockOnClick.mock.calls[0][1](null);
-  expect(wrapper.state().state).toBe("success");
-  expect(result).toBe(null);
-});
-
-it("onClick calls updates state and calls onClick with error", () => {
-  jest.useFakeTimers();
-  const mockOnClick = jest.fn();
-  const wrapper = shallow(generateComponent({
-    onClick: mockOnClick,
-  }));
-  wrapper.instance().onClick("e");
-  expect(wrapper.state().state).toBe("loading");
-  expect(mockOnClick).toHaveBeenCalledTimes(1);
-  expect(mockOnClick.mock.calls[0][0]).toBe("e");
-
-  // callback without error
-  mockOnClick.mock.calls[0][1](true);
-  expect(wrapper.state().state).toBe("error");
-  jest.runAllTimers();
-  expect(wrapper.state().state).toBe("default");
-});
+
+  it("renders with props", () => {
+    const wrapper = shallow(generateComponent());
+    expect(shallowToJson(wrapper)).toMatchSnapshot();
+  });
+
+  it("onClick calls updates state and calls onClick", () => {
+    const mockOnClick = jest.fn();
+    const wrapper = shallow(generateComponent({
+      onClick: mockOnClick,
+    }));
+    wrapper.instance().onClick("e");
```

Comment:
One thing we could do since we're already in here is simulate an actual click on the rendered button with something like `wrapper.find("button[type='submit']").simulate("click")` which would even test that the button is properly passed the onClick handler. I think that's also stylistically better, since it doesn't require us reaching into the component's hidden functions. Just the user interaction.